cypress/integration/unit_tests/run_test.js

```diff
@@ -0,0 +1,48 @@
+import * as Error from '../../../docs/error.js';
+import * as Resources from '../../../docs/resources.js';
+import {setScorePromiseAndReturnAssetName} from '../../../docs/run.js';
+import {loadScriptsBeforeForUnitTests} from '../../support/script_loader.js';
+
+describe('Unit test for run.js', () => {
+  loadScriptsBeforeForUnitTests('ee');
+  let errorStub;
+  beforeEach(() => errorStub = cy.stub(Error, 'showError'));
+
+  it('Score asset present', () => {
+    const assetName = 'TIGER/2018/States';
+    cy.stub(Resources, 'getScoreAsset').returns(assetName);
+    const backupStub = cy.stub(Resources, 'getBackupScoreAsset');
+    cy.wrap(setScorePromiseAndReturnAssetName()).then((result) => {
+      expect(result).to.equal(assetName);
+      expect(backupStub).to.not.be.called;
+      expect(errorStub).to.not.be.called;
+    });
+  });
+
+  it('Score asset not present, but backup is', () => {
+    cy.stub(Resources, 'getScoreAsset')
+        .returns('nonexistent/feature/collection');
+    const backupName = 'TIGER/2018/States';
+    cy.stub(Resources, 'getBackupScoreAsset').returns(backupName);
+    cy.wrap(setScorePromiseAndReturnAssetName()).then((result) => {
+      expect(result).to.equal(backupName);
+      expect(errorStub).to.be.calledOnce;
+    });
+  });
+
+
+  it('Neither asset present', () => {
+    cy.stub(Resources, 'getScoreAsset')
+        .returns('nonexistent/feature/collection');
+    cy.stub(Resources, 'getBackupScoreAsset').returns('another/bad/asset');
+    const promise = setScorePromiseAndReturnAssetName().then(
+        (result) => {
+          throw new Error('unexpected: ' + result);
```

Comment:
Do we want to throw an error here? Can we just assert result is null or something like that?